libtenzir/builtins/operators/time.cpp

```diff
@@ -0,0 +1,567 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#define _XOPEN_SOURCE 700 // for strptime
+
+#include <tenzir/argument_parser.hpp>
+#include <tenzir/arrow_table_slice.hpp>
+#include <tenzir/operator_control_plane.hpp>
+#include <tenzir/plugin.hpp>
+#include <tenzir/series_builder.hpp>
+
+// These are included in C++20, but not implemented (fully) in our stdlibs
+// These provide way better APIs compared to POSIX, that actually work
+#include <arrow/vendored/datetime/date.h>
+#include <arrow/vendored/datetime/tz.h>
+
+#include <ctime>
+
+namespace date = arrow_vendored::date;
+
+namespace tenzir::plugins::time {
+
+namespace {
+
+constexpr auto docs = "https://docs.tenzir.com/docs/next/formats/time";
+
+std::string strerror_nonbroken(int errnum) {
+  // strerror is not thread safe, strerror_r is
+  std::string buf(64, '\0');
+  ::strerror_r(errnum, buf.data(), buf.size());
+  return buf;
+}
```

Comment:
We also have `detail::describe_errno`, which does some additional error handling and resizing.